### ISSUE TITLE
Steam segfault when starting SteamVR 1487886215

### ISSUE THREAD

**Issue by ChristophHaag** [ValveSoftware/SteamVR-for-Linux#15]
Archlinux, RX 480

When started with `STEAM_RUNTIME=0 steam` it doesn't crash. With the runtime enabled, steam crashes when start steamvr, but vrmonitor and vrcompositor keep running and working.

steamvr system report
https://gist.github.com/ChristophHaag/e5ba90319a9d706a037a56e8b853a591

backtrace:
```
Thread 15 "CIPCServer::Thr" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0xe8febb40 (LWP 15818)]
0xf7c8b3a5 in pthread_join () from /usr/lib32/libpthread.so.0
(gdb) bt
#0  0xf7c8b3a5 in pthread_join () from /usr/lib32/libpthread.so.0
#1  0xd5c21431 in vortex::CVortexThread::~CVortexThread() ()
   from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#2  0xd5c1d30f in vortex::CLighthouseUsbThread::~CLighthouseUsbThread() ()
   from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#3  0xd5c18b01 in vortex::CLighthouseThreadBase::Stop() ()
   from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#4  0xd5c18c90 in vortex::CLighthouseInput::Stop() () from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#5  0xd5c18d91 in vortex::CLighthouseInput::CConnection::Disconnect() ()
   from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#6  0xd5c18ede in vortex::CLighthouseInput::CConnection::~CConnection() ()
   from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#7  0xd5b4121f in CWatchdogConnectionMonitor::~CWatchdogConnectionMonitor() ()
   from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#8  0xd5b40a39 in CWatchdogDriver_Lighthouse::Cleanup() ()
   from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/drivers/lighthouse/bin/linux32/driver_lighthouse.so
#9  0xd68b8c84 in CWatchdogTrackedDeviceProvider_Latest::Cleanup() () from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/bin/vrclient.so
#10 0xd686195a in CDriverManager::CleanupDriver(CDriverManager::HmdDriver_t*) () from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/bin/vrclient.so
#11 0xd686407f in CDriverManager::Cleanup() () from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/bin/vrclient.so
#12 0xd679f470 in CHmdSystemLatest::Cleanup() () from /home/chris/.local/share/Steam/SteamApps/common/SteamVR/bin/vrclient.so
#13 0xe8e330c4 in VR_ShutdownInternal () from /home/chris/.local/share/Steam/ubuntu12_32/libopenvr_api.so
#14 0xea067702 in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
---Type <return> to continue, or q <return> to quit---
#15 0xe9decbb4 in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#16 0xe9c24383 in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#17 0xe9ce490e in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#18 0xe9727458 in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#19 0xe9ce4008 in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#20 0xe9725f42 in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#21 0xe972871a in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#22 0xe9729546 in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/steamclient.so
#23 0xeff15e30 in SteamThreadTools::CThread::ThreadExceptionWrapper(void*) () from /home/chris/.local/share/Steam/ubuntu12_32/libtier0_s.so
#24 0xeff1421b in ?? () from /home/chris/.local/share/Steam/ubuntu12_32/libtier0_s.so
#25 0xeff144ef in CatchAndWriteMiniDumpExForVoidPtrFn () from /home/chris/.local/share/Steam/ubuntu12_32/libtier0_s.so
#26 0xeff14541 in CatchAndWriteMiniDumpForVoidPtrFn () from /home/chris/.local/share/Steam/ubuntu12_32/libtier0_s.so
#27 0xeff18ac1 in SteamThreadTools::CThread::ThreadProc(void*) () from /home/chris/.local/share/Steam/ubuntu12_32/libtier0_s.so
#28 0xf7c8a301 in start_thread () from /usr/lib32/libpthread.so.0
#29 0xf7bb446e in clone () from /usr/lib32/libc.so.6
```

**Comment by tizzir** [comment]
I also seem to be having the same issue.

Although I can't be sure that it's the cause, the issue began for me after updating SteamVR to the 1487886215 beta.

Also worth noting is the fact that I can launch the Dota 2 VR hub and Destinations separate from SteamVR and still have them work, minus the vr dashboard and other things I suspect are provided by SteamVR.

**Comment by tizzir** [comment]
On the first launch after updating  SteamVR to version 1487904548 Steam didn't crash. However, every attempt since still crashes Steam.

This could have just been a fluke, but either way at one point I did have both Steam and SteamVR running. (until Steam crashed afterwards, when I tried to run Portal 2 in VR theatre mode.)

**Comment by Cheeseness** [comment]
I'm still experiencing this with 1487904548 as well.

**Comment by michaellarabel** [comment]
I have been experiencing similar behavior to @tizzir where the first time worked but subsequent runs of SteamVR fail. rm -rf ~/.steam/, re-setting up Steam, opting into the Steam and SteamVR betas, then again works for only one time before the same behavior happens. On an Ubuntu 16.10 box.

**Comment by ryantux** [comment]
I have this problem also, first run is fine but a second run of SteamVR just hangs on "Preparing to launch SteamVR". I have to delete the installation folder as uninstall option wont work.

**Comment by Cheeseness** [comment]
> I have this problem also, first run is fine but a second run of SteamVR just hangs on "Preparing to launch SteamVR". I have to delete the installation folder as uninstall option wont work.

Do you reliably get this every time? If so, can you try turning off Steam cloud support before running it for the first time and see if the problem persists? To do this, right click on the game in your Steam library, select Properties, then in the Updates tab, uncheck the "Enable Steam Cloud synchronization for SteamVR" box.

**Comment by anoadragon453** [comment]
Apparently running with Steam stable client (opting out of the beta) seems to fix the issue, though I agree this is a poor workaround.

**Comment by ryantux** [comment]
@Cheeseness, it happens every time on second launch after a clean install of SteamVr.
Disabling the steam cloud option does fix this so long as it's done before running it for first time as you say.
Thanks

**Comment by Cheeseness** [comment]
@ryantux Thanks for confirming my suspicions! Smells like maybe the Steam client gets stuck thinking it's uploading something to to the cloud when SteamVR crashes (might be crashing on exit if you're not noticing crashes?)

It might be worth making a separate issue for this problem, though I'm not certain whether the SteamVR bug tracker or the Steam client bug tracker is the best place for it to go.

**Comment by valentindavid** [comment]
I have the same SIGSEGV in Steam client as OP. Same stack trace. The solution to use STEAM_RUNTIME=0 worked. But then SteamVR would SIGSEGV in vrcompositor, the stacktrace pointed to some Vulkan initialization, even though I had the Vulkan loader 1.0.39 like the Steam runtime. I tried to trick the launch script of SteamVR to force using the Steam runtime, but that did not help. The solution I have found was to copy .steam/steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libvulkan.so.1 into a separate directory and start the Steam client with STEAM_RUNTIME=0 and LD_LIBRARY_PATH pointing to that directory. Now it works. Both Steam client and SteamVR can start without crashing. However Steam crashes when I stop SteamVR. But it is less annoying.
